packages/ketcher-core/src/application/editor/modes/SequenceMode.ts

```diff
@@ -835,20 +835,51 @@ export class SequenceMode extends BaseMode {
     const chainsCollection = ChainsCollection.fromMonomers([
       ...drawingEntitiesManager.monomers.values(),
     ]);
+
+    const currentSequence = SequenceRenderer.currentChain;
+
+    const currentSequenceHasPhosphate = currentSequence?.lastNonEmptyNode
+      ? this.hasPhosphateAtEndOfSequence(
+          currentSequence.lastNonEmptyNode.monomer,
+        )
+      : false;
+
+    const newSequenceHasPhosphate = chainsCollection.lastNode
+      ? this.hasPhosphateAtEndOfSequence(chainsCollection.lastNode.monomer)
+      : false;
```

Comment:
It seems that you may omit these ternaries here by passing monomer directly via conditional chaining, even if it's `undefined`/`null`, `hasPhosphateAtEndOfSequence` will return you false, if you apply conditional chaining there as well